### PR TITLE
Create test to object declaration scope

### PR DESCRIPTION
**Pull request by bloderxd**
This PR proposes the creation of tests to scopes starting with `ObjectDeclaration`, these tests could help on documentation building since we can get all scope property informations that we need.

### CODE REVIEW

File: compiler-plugin/src/main/kotlin/arrow/meta/phases/analysis/DefaultElementScope.kt [hunk 1]
```diff
@@ -93,12 +93,12 @@ class DefaultElementScope(project: Project) : ElementScope {
...
-  override val companionObject: Scope<KtObjectDeclaration>
-    get() = Scope(delegate.createCompanionObject())
-  override val String.companionObject: Scope<KtObjectDeclaration>
-    get() = Scope(delegate.createCompanionObject(trimMargin()))
+  override val String.`object`: ObjectDeclaration
+    get() = ObjectDeclaration(delegate.createObject(trimMargin()))
+  override val companionObject: ObjectDeclaration
+    get() = ObjectDeclaration(delegate.createCompanionObject())
+  override val String.companionObject: ObjectDeclaration
+    get() = ObjectDeclaration(delegate.createCompanionObject(trimMargin()))
```
Comment (ahinchman1) [inline compiler-plugin/src/main/kotlin/arrow/meta/phases/analysis/DefaultElementScope.kt]: good catch! Thanks for this

File: compiler-plugin/src/test/kotlin/arrow/meta/quotes/scope/ObjectDeclarationTest.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+                         | //metadebug
+                         | @Deprecated("Test") object Test {
+                         |   fun test() { println("Test") }
+                         |   fun test2() { println("Test2") }
+                         | }""".source
+  
+  @Test
+  fun `validate object declaration scope properties`() {
+    assertThis(CompilerTest(
+      config = { metaDependencies + addMetaPlugins(ScopeMetaPlugin()) },
```
Comment (ahinchman1) [inline compiler-plugin/src/test/kotlin/arrow/meta/quotes/scope/ObjectDeclarationTest.kt]: so awesome! Thank you for this! I'm going to leave a comment at the top of the review for easy access/docs

**Comment by ahinchman1** [comment]
One tip to discover all possible elements - use the IDEA debugging (thanks to @bloderxd) <3

Captura de Tela 2019-11-27 às 02 43 32

**Comment by raulraja** [comment]
thanks @bloderxd this will help us test the quote system!

**Pull request merged**